app/models/vendor_code.rb

```diff
@@ -7,11 +7,8 @@ class VendorCode < ActiveRecord::Base
                    length: { maximum: 255, message: "longer than 255 characters" }
 	
 	
-  def assign_to(current_user)
-    self.update_attributes(user: current_user, user_name: current_user.name, email: current_user.email)
-    vendor = self.vendor
-    vendor.update_attribute(:usedCodes, vendor.usedCodes + 1)
-    vendor.update_attribute(:unclaimCodes, vendor.unclaimCodes - 1)
+  def associate_with(user)
+    self.update(user_id: user.id, user_name: user.name, email: user.email)
   end  
```

Comment:
Hi, Oleg
It was done intentionally.
The only code we can assign to the user is redeemify code.
To see the same offers from vendors of customers page we associate vendor codes with the user until they will be redeemed.
To illustrate on an example:
1. log in as vendor;
2. upload codes;
3. log in as a user;
4. enter valid redeemify code;
5. switch to the vendor account.
The number of vendor codes has been reduced by one, whereas it should remain unchanged.